### PR TITLE
Support `preload` option in `Worker`

### PR DESCRIPTION
**Pull request by Jarred-Sumner**
### What does this PR do?

Support `preload` option in `Worker`

Fixes https://github.com/oven-sh/bun/issues/15044

### How did you verify your code works?

Tests

### CODE REVIEW

**Comment by robobun** [comment]
@Jarred-Sumner, your commit 96e717082a79fce8cfc010e759a5eecc833bb572 is building: [`#5787`](https://buildkite.com/bun/bun/builds/5787)

File: src/bun.js/bindings/webcore/Worker.cpp [hunk 1]
```diff
@@ -160,6 +162,16 @@ ExceptionOr<Ref<Worker>> Worker::create(ScriptExecutionContext& context, const S
 
     Vector<String>* argv = worker->m_options.bun.argv.get();
     Vector<String>* execArgv = worker->m_options.bun.execArgv.get();
+    Vector<String>* preloadModuleStrings = &worker->m_options.bun.preloadModules;
+    Vector<BunString> preloadModules;
+    preloadModules.reserveInitialCapacity(preloadModuleStrings->size());
+    for (auto& str : *preloadModuleStrings) {
+        if (str.startsWith("file://"_s)) {
+            str = WTF::URL(str).fileSystemPath();
```
Comment (paperclover) [inline src/bun.js/bindings/webcore/Worker.cpp]: url could start with file:// but be invalid
```
WTF::URL url(str);
if (url.isValid()) str = url.fileSystemPath()
```

File: src/bun.js/web_worker.zig [hunk 2]
```diff
@@ -90,6 +179,8 @@ pub const WebWorker = struct {
         argv_len: u32,
         execArgv_ptr: ?[*]WTFStringImpl,
         execArgv_len: u32,
+        preload_modules_ptr: ?[*]bun.String,
```
Comment (paperclover) [inline src/bun.js/web_worker.zig:182]: c doesn't have a way to pass a non-null garbage pointer? yikes.

**Review by paperclover — CHANGES_REQUESTED** [review]
one nitpick

**Comment by Jarred-Sumner** [comment]
Fixed

**Pull request merged**